### ISSUE TITLE
Do toggle buttons not require the aria-pressed attribute in their markup?

### ISSUE THREAD

**Issue by sivakusayan** [w3c/wcag#2038]
Sorry if this was already brought up - I did a search and couldn't find anything in the issue list.

In the [documentation for toggle buttons](https://www.w3.org/TR/wai-aria-practices-1.1/#button), the following passage seems to imply that toggle buttons don't need `aria-pressed` as long as their caption changes:

> **Important:** it is critical the label on a toggle does not change when its state changes. In this example, when the pressed state is true, the label remains "Mute" so a screen reader would say something like "Mute toggle button pressed". Alternatively, if the design were to call for the button label to change from "Mute" to "Unmute," the aria-pressed attribute would not be needed.

Does this mean that buttons where the caption changes should be treated as toggle buttons? Doing a quick ~5 minute experiment shows that both JAWs and NVDA only recognize buttons as a toggle button if they have a defined `aria-pressed` attribute. Furthermore, if a toggle button instead omits the `aria-pressed` attribute and just changes its caption, neither JAWs nor NVDA will read out the new caption in Chrome. In Firefox, NVDA will read out the caption, but not JAWs.[ JSFiddle](https://jsfiddle.net/b143tLan/1/)

So I'm wondering what expected behavior should be here - is it really okay for a button to change its caption to signify state, and if so should browsers and assistive technology detect that and notify users of the updated button caption? Or should toggle buttons require the `aria-pressed` attribute?

Edit: It turns out that [this part of the specification](https://www.w3.org/TR/wai-aria-practices-1.1/#wai-aria-roles-states-and-properties-3) calls out that toggle buttons need the aria-pressed attribute. So it feels like we'll need to update documentation in at least one of the two quoted places either way.

**Comment by patrickhlauke** [comment]
Imagine a button that changes its actual text/accessible name from "Play" to "Pause". if you now add `aria-pressed`, you start adding an extra layer of confusion. "Play, not pressed", "Pause, pressed" / "Play, pressed", "Pause, not pressed". user has to puzzle over "so does it mean if it's currently playing or paused? and what happens if i press it?

the fact that many current browser/AT combos don't announce a change of text if the currently focused element is randomly changed under the user's feet is a shortcoming, but can be worked around (e.g. adding `aria-live="polite"` to the button)

**Comment by patrickhlauke** [comment]
also, suggest that this isn't the right repo to ask this - should instead ask over on https://github.com/w3c/aria-practices/issues/

**Comment by patrickhlauke** [comment]
> is it really okay for a button to change its caption to signify state

it's not so much that it changes _state_, but it changes _behaviour_/meaning/purpose. using the play/pause example, when nothing is playing, the button is "Play". once playing, the purpose of the button turns to "Pause"

**Comment by sivakusayan** [comment]
> the fact that many current browser/AT combos don't announce a change of text if the currently focused element is randomly changed under the user's feet is a shortcoming, but can be worked around (e.g. adding `aria-live="polite"` to the button)

This was really the thing I was wondering - we do technically have workarounds, but I was wondering whether this should be considered a bug with the browser/AT or if this is just weird wording of the spec. Although, now that I searched that repo, I came upon [this link](https://github.com/w3c/aria-practices/issues/121) which shows this was intentional behavior, albeit just not widely implemented.

Thanks for your help in any case!

**Comment by patrickhlauke** [comment]
essentially, when you're re-labelling the control, it's not a "toggle" in the `aria-pressed` sense anymore. colloquially you might still call it a toggle, but a toggle has the same name and just changes states

**Comment by JAWS-test** [comment]
WCAG SC 1.3.1:

> Information, structure, and relationships conveyed through presentation can be *programmatically determined or are available in text.*

This means for the toggle button that you have two options:
*  programmatically determined: use aria-pressed
* are available in text: change the label

Now, if you know that a toggle button with label change does not work well due to the AT, then you should obviously avoid it to satisfy WCAG 5.2.4: https://www.w3.org/TR/WCAG21/#cc4
